pkg/fswatcher/fswatcher_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2021 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fswatcher
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zaptest/observer"
+)
+
+func TestFSWatcherAddFiles(t *testing.T) {
+	w, err := NewFSWatcher([]string{"foo"}, nil, nil)
+	assert.Error(t, err)
+	assert.IsType(t, &FSWatcher{}, w)
+
+	w, err = NewFSWatcher([]string{"../../cmd/query/app/fixture/ui-config.json"}, nil, nil)
+	assert.NoError(t, err)
+	assert.IsType(t, &FSWatcher{}, w)
+
+	w, err = NewFSWatcher([]string{"../../cmd/query/app/fixture/ui-config.json", "foo"}, nil, nil)
+	assert.Error(t, err)
+	assert.IsType(t, &FSWatcher{}, w)
+
+	w, err = NewFSWatcher([]string{"../../cmd/query/app/fixture/static/asset.txt", "../../cmd/query/app/fixture/ui-config.json"}, nil, nil)
+	assert.NoError(t, err)
+	assert.IsType(t, &FSWatcher{}, w)
+
+	err = w.Close()
+	assert.NoError(t, err)
+}
+
+func TestFSWatcherWatchFileChangeAndRemove(t *testing.T) {
+	testFile, err := os.Create("test.doc")
+	require.NoError(t, err)
+
+	_, err = testFile.WriteString("test content")
+	require.NoError(t, err)
+
+	zcore, logObserver := observer.New(zapcore.InfoLevel)
+	logger := zap.New(zcore)
+
+	onChange := func() {
+		logger.Info("Content changed")
```

Comment:
Nit: better test signal would be to read the file content and include it in the log, that way you can be sure the write log line is matched against